### ISSUE TITLE
Investigate shape noise in mock_data

### ISSUE THREAD

**Issue by combet** [LSSTDESC/CLMM#304]
At the moment in mock data, shape noise is implemented as a random component (from a gaussian distribution) added to the reduced tangential shear. Ellipticities are then computed from this 'noisy' `g_t`.

Another approach (maybe equivalent, but I'm not sure), more representative of what actually happens would be to:
- draw intrinsic ellipticities `e1_intrinsic` and `e2_intrinsic` from gaussian distributions, between -1 and 1;
- compute the reduced tangential shear `g_t` and convergence `kappa` due to the cluster;
- compute the lensed ellipticities `e1` and `e2` from the above, using the transformation from Schneider
![Screenshot 2020-09-28 at 09 30 37](https://user-images.githubusercontent.com/20227809/94403136-4542c980-016d-11eb-87ea-8ca00613529c.png)

The transformation from source to lensed ellipticities (eq. above (right), |g|<1 case,  but with  g --> -g), is already implemened in the function `compute_lensed_ellipticities` in `utils.py`.

**Comment by combet** [comment]
I've very naively implemented the above in `mock_data.py`.  

Even without asking for shape noise, the `Example1...` notebook does not perform the fit properly anymore. I think this indicates either a signa problem with the lines:
- `g1 = -gamt * np.cos(2*galaxy_catalog['posangle']) + gamx * np.sin(2*galaxy_catalog['posangle'])` and  
`g2 = -gamt * np.sin(2*galaxy_catalog['posangle']) - gamx * np.sin(2*galaxy_catalog['posangle'])`
 
or with

- `galaxy_catalog['e1'],galaxy_catalog['e2'] = compute_lensed_ellipticity(e1_intrinsic, e2_intrinsic, g1, g2, kappa)`

This needs checking!

**Comment by combet** [comment]
@vitenti identified the issue. In`compute_lensed_ellipticity`, the shear1 and shear2 components correspond to the shear, not the reduced shear. However, in the `_generate_galaxy_catalog` function of `mock_data`, I was using the reduced tangential shear (which is what we wanted with the former implementation of the shape noise).
- this has been corrected; 
- `g1` and `g2` have been renamed `gam1` and `gam2`, to be reminiscent of gamt and gamx that correspond to the (not reduced) shear;
- the docstrings of `compute_lensed_ellipticity` has been made more explicit to avoid further confusion.
- I've checked that all the notebooks relying on mock data generation now produce the expected results.
